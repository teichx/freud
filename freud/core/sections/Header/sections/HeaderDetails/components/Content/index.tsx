import { ContentAppearance } from './ContentAppearance';
import { ContentDefault } from './ContentDefault';
import { ContentLanguage } from './ContentLanguage';
import { ContentProps, ContentVariationProps } from './types';

const contentVariation: ContentVariationProps = {
  close: () => null,
  default: ContentDefault,
  language: ContentLanguage,
  appearance: ContentAppearance,
};

export const Content = ({ toContent, content }: ContentProps) => {
  const ContentComponent = contentVariation[content];

  return <ContentComponent toContent={toContent} />;
};
