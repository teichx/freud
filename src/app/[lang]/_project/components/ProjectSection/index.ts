export * from './ProjectSection';
export * from './ProjectFirstSection';
export * from './ProjectSectionImage';
export * from './ProjectSectionWrapper';
export * from './types';
