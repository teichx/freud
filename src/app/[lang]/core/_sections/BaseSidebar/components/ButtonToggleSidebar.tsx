import { ChevronLeftIcon, ChevronRightIcon } from '@chakra-ui/icons';
import { Button } from '@chakra-ui/react';

import { useScopedI18n } from '~/i18n/client';

import { useSidebarContext } from '../context';

export const ButtonToggleSidebar = () => {
  const { isOpen, onToggle } = useSidebarContext();
  const t = useScopedI18n('components.sidebar');

  return (
    <Button
      size='sm'
      leftIcon={isOpen ? <ChevronLeftIcon /> : undefined}
      rightIcon={isOpen ? undefined : <ChevronRightIcon />}
      onClick={onToggle}
      aria-label='Sidebar aria label'
    >
      {t(isOpen ? 'close' : 'open')}
    </Button>
  );
};
