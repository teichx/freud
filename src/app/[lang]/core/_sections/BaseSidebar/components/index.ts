export * from './ButtonToggleSidebar';
export * from './SidebarButton';
export * from './SidebarWrapper';
