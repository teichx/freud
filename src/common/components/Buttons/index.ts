export * from './variant';
export * from './types';

import {
  BackButton,
  CancelButton,
  LinkButton,
  SaveButton,
  Microphone,
} from './variant';

export const Buttons = {
  Back: BackButton,
  Cancel: CancelButton,
  Link: LinkButton,
  Save: SaveButton,
  Microphone: Microphone,
};
