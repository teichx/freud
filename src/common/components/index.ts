export * from './Avatar';
export * from './Breadcrumb';
export * from './Buttons';
export * from './DataModal';
export * from './DataModal';
export * from './DataTable';
export * from './Display';
export * from './Form';
export * from './Input';
export * from './GoogleLogin';
export * from './InfiniteScroll';
export * from './Loader';
export * from './Logo';
export * from './Meta';
export * from './ReadOnlyText';
export * from './Section';
export * from './Select';
export * from './SkeletonContextLoader';
export * from './TextHighlight';
export * from './ToggleTheme';
export * from './TooltipComponent';
export * from './TooltipDate';
