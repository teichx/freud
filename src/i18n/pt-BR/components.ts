export const components = {
  buttons: {
    microphone: {
      toDisable: 'Desligar microfone',
      toEnable: 'Ativar microfone',
      helperText: {
        editOnlyNotListening:
          'O campo não pode ser editado enquanto o microfone estiver ligado',
      },
    },
    login: 'Fazer login',
    access: 'Entrar como {name}',
    logout: 'Deslogar',
    back: 'Voltar',
    cancel: 'Cancelar',
    save: 'Salvar',
  },
  toggleTheme: {
    label: 'Usar tema {appearance}',
    light: 'claro',
    dark: 'escuro',
  },
  form: {
    disabledByPristine: 'Não há alterações a serem salvas',
    disabledByHasValidationErrors: 'Há dados inválidos no formulário',
  },
  dataTable: {
    noData: 'Sem dados por aqui',
    loading: 'Carregando linhas',
    pagination: 'Página {page} de {pageCount}',
    rowsPerPage: 'Linhas por página:',
    pageHint: {
      first: 'Primeira página',
      next: 'Próxima página',
      before: 'Página anterior',
      last: 'Última página',
    },
  },
  select: {
    empty: {
      unfilled: 'Sem mais items',
      filled: 'Não há outros resultados para "{inputValue}"',
    },
    placeholder: {
      unique: 'Selecione uma opção',
      multiple: 'Selecione as opções',
    },
  },
  infiniteScroll: {
    endOfList: 'Sem mais items',
  },
  sidebar: {
    open: 'Abrir barra lateral',
    close: 'Fechar barra lateral',
  },
} as const;
