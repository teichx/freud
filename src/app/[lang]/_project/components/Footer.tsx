'use server';
import {
  Box,
  Container,
  Icon,
  IconButton,
  Text,
  HStack,
} from '@chakra-ui/react';
import { BsGithub } from 'react-icons/bs';
import { GiBrazil } from 'react-icons/gi';

import { LinkButton } from '~/common/components/Buttons/variant/LinkButton';
import { TooltipComponent } from '~/common/components/TooltipComponent';
import { getScopedI18n } from '~/i18n/server';

export const Footer = async () => {
  const t = await getScopedI18n('project');

  return (
    <Box
      w='100%'
      sx={{
        bg: 'book.darkBlue.500',
        color: 'gray.200',
        _dark: {
          bg: 'book.desertSun.900',
        },
      }}
    >
      <Container w='100%' maxW='container.xl' fontSize='sm'>
        <HStack
          justifyContent='center'
          py='2'
          columnGap='1'
          sx={{
            '> *': {
              marginInlineStart: '0 !important',
            },
          }}
        >
          <Text m='0'>{t('author.createdOn')}</Text>

          <TooltipComponent label={t('author.brazil')}>
            <Box
              display='flex'
              alignItems='center'
              sx={{
                color: 'book.desertSun.500',
                _dark: {
                  color: 'white',
                },
              }}
            >
              <Icon viewBox='0 0 14 14'>
                <GiBrazil />
              </Icon>
            </Box>
          </TooltipComponent>

          <Text>{t('author.by')}</Text>

          <LinkButton
            p='0'
            height='auto'
            color='inherit'
            variant='link'
            fontSize='inherit'
            fontWeight='inherit'
            linkProps={{
              target: '_blank',
            }}
            href='https://github.com/teichx'
          >
            {t('author.teichx')}
          </LinkButton>

          <IconButton
            size='xs'
            rounded='full'
            sx={{
              color: 'black',
              _dark: {
                color: 'white',
              },
            }}
            aria-label='https://github.com/teichx/freud'
            icon={
              <Icon viewBox='0 0 16 16' fontSize='md'>
                <BsGithub />
              </Icon>
            }
          />
        </HStack>
      </Container>
    </Box>
  );
};
