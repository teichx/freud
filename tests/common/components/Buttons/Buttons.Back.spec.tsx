import { render } from 'test-utils';

import { Buttons } from '~/common/components/Buttons';

describe('component.Buttons', () => {
  describe('buttons.Back', () => {
    it('default', () => {
      expect.assertions(2);

      const { asFragment, getByText } = render(<Buttons.Back />);

      expect(asFragment()).toMatchSnapshot();
      expect(getByText('translations.words.back')).toBeDefined();
    });

    it('with text', () => {
      expect.assertions(1);
      const text = '<current-text>';

      const { getByText } = render(<Buttons.Back text={text} />);

      expect(getByText(text)).toBeDefined();
    });

    it('with children', () => {
      expect.assertions(1);

      const text = '<current-children>';

      const { getByText } = render(<Buttons.Back>{text}</Buttons.Back>);

      expect(getByText(text)).toBeDefined();
    });
  });
});
