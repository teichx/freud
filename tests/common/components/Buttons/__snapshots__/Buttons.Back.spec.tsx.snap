// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.Buttons buttons.Back default 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-pmna7d"
    type="button"
  >
    words.back
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
